pages/api/auth/[...nextauth].ts

```diff
@@ -115,7 +115,22 @@ export const authOptions: NextAuthOptions = {
   adapter: PrismaAdapter(prisma),
   events: {
     async signIn({ user }) {
-      logEvent(user.id, { type: "User", id: user.id }, "UserSignIn");
+      if (!user.name || !user.email)
+        throw new Error(
+          "Could not produce the UserSignIn audit log because of missing name and/or email"
+        );
+
+      const internalUser = await getOrCreateUser(user.name, user.email, user.image);
+
+      if (internalUser === null)
```

Comment:
we would expect that if calling getOrCREATEUser we would never get a null..

Reply:
The main reason why it can return a `null` value is because of that `prismaErrors` function that catches specific database internal errors and may end up returning a default value which in our case is a `null`.
I can not remember why we did it this way instead of using a throw/catch pattern.
cc @bryan-robitaille